controller/src/main/java/org/apache/rocketmq/controller/impl/manager/ReplicasInfoManager.java

```diff
@@ -533,7 +533,7 @@ private void handleElectMaster(final ElectMasterEvent event) {
             } else {
                 // If new master was not elected, which means old master was shutdown and the newSyncStateSet list had no more replicas
                 // So we should delete old master, but retain newSyncStateSet list.
-                syncStateInfo.updateMasterInfo(null);
+                syncStateInfo.cleanMasterInfo();
```

Comment:
If the master's epoch is not increased and is set to null, it appears that the broker will not actively trigger a re election in the current implementation.If the master's epoch is not increased and is set to null, it appears that the broker will not actively trigger a re election in the current implementation.

Reply:
Ok, let me understand what you mean.
> it appears that the broker will not actively trigger a re election in the current implementation.

Because in current implementation, broker will send elect request proactive when the epoch increased and master is null. code as follow:
```java
if (newMasterEpoch > this.masterEpoch) {
      if (StringUtils.isNoneEmpty(newMasterAddress) && masterBrokerId != null) {
          ....
      } else {
          // In this case, the master in controller is null, try elect in controller, this will trigger the electMasterEvent in controller.
          brokerElect();
      }
  }
```
If the epoch not increased when the master shutdown, this step maybe never occur. But discontinuous incremental epochs can be very confusing.